backend/src/layer/data/database/mealplan_management.rs

```diff
@@ -333,6 +355,36 @@ impl PersistentMealplanManagementData {
         .execute(&self.pool)
         .await?;
 
+        if nutrition_data.is_some() {
+            let nut_data = nutrition_data.expect("If you see this, logic broke");
```

Comment:
The nice way to do this:
```suggestion
        if let Some(nut_data) = nutrition_data {
```

Reply:
Right. Thank you. I forgot that this worked in Rust. It's been a while...